orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -97,7 +97,7 @@ def __init__(self, module):
         self._export_again = False
         # TODO: This is incorrect when different layers may be in different devices
         self._device = next(module.parameters()).device
-        self._require_export = False
+        self._device_changed = False
```

Comment:
The only time export is required is when the model has never been exported before or if the device has changed. Is my understanding correct?
Do you think we could have other reasons in the future to require a complete export (maybe in that case we could keep the name the same as it was ```_require_export```?)

Reply:
We don't actually re-export the ONNX graph when the device changes. We just recreate the InferenceSession
This flag was poorly named